### PR TITLE
validate input/output options are objects

### LINKED ISSUE #2183: [suggestion] Input validation for JS api
Listing this as a suggestion since I don't consider it a bug, but mostly a quality-of-life change.

I'm experimenting rollup for the first time and ended up passing a string to the `output` option. 100% my fault!

```
  const bundle = await rollup.rollup({
    input: "./lib/index.ts",
    output: "./built/index.js",
```

That brought me the following error, which was enough for me to find my fault, but took some time.

```
TypeError: Cannot use 'in' operator to search for 'name' in ./built/index.js
    at deprecateToOutputOption (c:\Users\krueg\Desktop\wfcore\node_modules\rollup\dist\rollup.js:323:25)
```

Would be great if instead I got an error telling me something like "output key expects an object". I'd suggest passing the input through some validation lib like Yup (yup has cool TS interop, too).

Ty for great work!

### PR DESCRIPTION
**Pull request by sach2211**
This PR attempts to add validations mentioned in https://github.com/rollup/rollup/issues/2183

Adding checks to ensure input / output options are in form of an object. If not, throw a meaningful error message.

resolves #2183

### CODE REVIEW

File: src/rollup/index.ts [hunk 1]
```diff
@@ -93,6 +93,9 @@ function getInputOptions(rawInputOptions: GenericConfigObject): any {
 	if (!rawInputOptions) {
 		throw new Error('You must supply an options object to rollup');
 	}
+	if (typeof rawInputOptions !== 'object') {
+		throw new Error(`You must supply input options as an object, encountered ${typeof rawInputOptions}`);
```
Comment (guybedford) [inline src/rollup/index.ts]: This can be a `TypeError`.

File: src/rollup/index.ts [hunk 2]
```diff
@@ -504,6 +507,9 @@ function normalizeOutputOptions(
 	if (!rawOutputOptions) {
 		throw new Error('You must supply an options object');
 	}
+	if ( typeof rawOutputOptions !== 'object') {
+		throw new Error(`You must supply output options as an object, encountered ${typeof rawOutputOptions}`);
```
Comment (MKrupauskas) [inline src/rollup/index.ts]: also TypeError

**Comment by shellscape** [comment]
This seems like a good opportunity for encapsulating some of this type of validation. I've never been a fan of [ajv](https://www.npmjs.com/package/ajv) (it's overly verbose, hard to version, and the output is very ugly), but something like [joi](https://www.npmjs.com/package/joi) looks to have promise and is expressive.

**Comment by shellscape** [comment]
@sach2211 hey there. Thanks for all of your work on this PR so far. It looks like it needs to be rebased to the current master to keep it going. Would you be up for that, or would you prefer to close this one?

**Comment by shellscape** [comment]
Closing this one for now. @sach2211 if you'd like to pick it back up, please ping!

**Pull request closed without merging**